src/mds/Server.cc

```diff
@@ -559,6 +559,23 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
     return;
   }
 
+  if (session->auth_caps.rootsquash_in_caps() &&
```

Comment:
Wondering if I should also do this in `Server::handle_client_reconnect()`

Reply:
IMO this makes sense. 

Just in case when upgrading the ceph cluster, but the clients won't or not finished yet, then when the old clients could successfully reconnect to the servers.

Reply:
Done.